lib/glimmer/tk/toplevel_proxy.rb

```diff
@@ -32,19 +32,43 @@ class ToplevelProxy < WidgetProxy
       DEFAULT_HEIGHT = 0
       
       attr_reader :tk
-      attr_accessor :escapable
+      attr_accessor :escapable, :modal, :centered
       alias escapable? escapable
-      
+      alias modal? modal
+      alias centered? centered
+
       def post_add_content
-        center_within_screen unless @x || @y
+        center_within_screen if centered?
+
         if escapable?
            on('KeyPress') do |event|
-            if event.keysym == 'Escape'
-              grab_release
+            if event.state == 0 && event.keysym == 'Escape'
+              tk.grab_release
               destroy
             end
           end
         end
+
+        if is_a?(Glimmer::Tk::ToplevelProxy) && modal?
```

Comment:
I am not sure why you are checking if `is_a?(Glimmer::Tk::ToplevelProxy)`. The code is inside an instance method of `Glimmer::Tk::ToplevelProxy`, so `self` will always have `is_a?(Glimmer::Tk::ToplevelProxy)` as `true`.